src/lamp_py/bus_performance_manager/events_tm.py

```diff
@@ -132,9 +134,10 @@ def generate_tm_events(tm_files: List[str]) -> pl.DataFrame:
         )
         .join(tm_geo_nodes, on="GEO_NODE_ID", how="left", coalesce=True)
         .join(tm_time_points, on="TIME_POINT_ID", how="left", coalesce=True)
-    ).sort(by="PATTERN_ID")
+    ).sort(by=["TRIP_SERIAL_NUMBER", "PATTERN_ID", "PATTERN_GEO_NODE_SEQ"])
```

Comment:
nit: sort is not necessary if this is just use an input to an min/max agg function.